File: TakeHome28072022/Shop/src/main/java/com/practice/shop/entity/Product.java
package com.practice.shop.entity;
import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
@Entity
@Table(name = "product")
public class Product {
	@Id
	@Column(name = "id")
	private Integer id;
	@Column(name = "sku")
	private String sku;
	@Column(name = "name")
	private String name;
	@Column(name = "description")
	private String description;
	@Column(name = "unitprice")
	private double unitprice;
	@Column(name = "imageurl")
	private String imageurl;
	@Column(name = "active")
	private byte active;
	@Column(name = "unitinstock")
	private Integer unitinstock;
	@Column(name = "datecreated")
	private String datecreated; 
	@Column(name = "lastupdated")
	private String lastupdated;
	@Column(name = "categoryid")
	private Integer categoryid;
	public Integer getId() {
		return id;
	}
	public void setId(Integer id) {
		this.id = id;
	}
	public String getSku() {
		return sku;
	}
	public void setSku(String sku) {
		this.sku = sku;
	}
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public String getDescription() {
		return description;
	}
	public void setDescription(String description) {
		this.description = description;
	}
	public double getUnitprice() {
		return unitprice;
	}
	public void setUnitprice(double unitprice) {
		this.unitprice = unitprice;
	}
	public String getImageurl() {
		return imageurl;
	}
	public void setImageurl(String imageurl) {
		this.imageurl = imageurl;
	}
	public byte getActive() {
		return active;
	}
	public void setActive(byte active) {
		this.active = active;
	}
	public Integer getUnitinstock() {
		return unitinstock;
	}
	public void setUnitinstock(Integer unitinstock) {
		this.unitinstock = unitinstock;
	}
	public String getDatecreated() {
		return datecreated;
	}
	public void setDatecreated(String datecreated) {
		this.datecreated = datecreated;
	}
	public String getLastupdated() {
		return lastupdated;
	}
	public void setLastupdated(String lastupdated) {
		this.lastupdated = lastupdated;
	}
	public Integer getCategoryid() {
		return categoryid;
	}
	public void setCategoryid(Integer categoryid) {
		this.categoryid = categoryid;
	}
	public Product(Integer id, String sku, String name, String description, double unitprice, String imageurl, byte active,
			Integer unitinstock, String datecreated, String lastupdated, Integer categoryid) {
		super();
		this.id = id;
		this.sku = sku;
		this.name = name;
		this.description = description;
		this.unitprice = unitprice;
		this.imageurl = imageurl;
		this.active = active;
		this.unitinstock = unitinstock;
		this.datecreated = datecreated;
		this.lastupdated = lastupdated;
		this.categoryid = categoryid;
	}
	public Product() {
		
	}
	
	

}

File: GuidedAssignment01082022/Shop src Angular/app/common/product.ts
// "id" : 1003,
//       "sku" : "MDN-5003",
//       "name" : "AC",
//       "description" : "This is a air conditioner",
//       "unitprice" : 35000.0,
//       "imageurl" : "https://os-wordpress-media.s3.ap-south-1.amazonaws.com/blog/wp-content/uploads/2020/11/24212247/Air-Conditioner-Buying-Guide.jpg",
//       "active" : 0,
//       "unitinstock" : 15,
//       "datecreated" : "2019-12-24T18:30:00.000+00:00",
//       "lastupdated" : "2022-06-10T18:30:00.000+00:00",
//       "categoryid" : 3,
export class Product {
  constructor(
    public id : number,
    public sku : string,
    public name : string,
    public description : string,
    public unitprice : number,
    public imageurl : string,
    public active : number,
    public unitinstock : number,
    public datecreated : string,
    public lastupdated : string,
    public categoryid : number
  ){}
}
